src/main/java/codeu/model/data/User.java

```diff
@@ -141,9 +141,11 @@ public void setPals(List<String> pals) {
         return pals;
     }
 
-    /** Adds a pal to the list of this User's pals. */
+    /** Adds a pal to the list of this User's pals as long as the pal is not this User. */
     public void addPal(String name) {
```

Comment:
validate input?

Reply:
What if we've already added this user as a pal? Will s/he be a double pal?

Reply:
yes that is a good point! I added statements to check that the name passed in is a valid user (with only characters that were allowed in LoginServlet.java) and that the username is a registered user and that the name is not a pal already.